### PR TITLE
pkg/daemon: roll back pending config on error

### PR DESCRIPTION
**Pull request by runcom**
If we fail something after we write files to disk for desired config we
now roll back to files for current config.
In case we fail _after_ writing the pending config file, we still roll
back files but we never remove the pending config.
I was debugging
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/22653/pull-ci-openshift-origin-master-e2e-aws-upgrade/89
and what it looks like it's happening is that we wrote the pending
config but we then errored, rolling back files but leaving that pending
config file around which made the MCD restart fail (also hiding the real
failure error on reboot).

### CODE REVIEW

**Comment by runcom** [comment]
Similar job failure here https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-network-operator/156/pull-ci-openshift-cluster-network-operator-master-e2e-aws-upgrade/25

**Comment by runcom** [comment]
Ok, I now understand what's happening (we fail to set Done and when we retry we're busted):

```
message: "unable to update node "&Node{ObjectMeta:k8s_io_apimachinery_pkg_apis_meta_v1.ObjectMeta{Name:,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,},Spec:NodeSpec{PodCIDR:,DoNotUse_ExternalID:,ProviderID:,Unschedulable:false,Taints:[],ConfigSource:nil,},Status:NodeStatus{Capacity:ResourceList{},Allocatable:ResourceList{},Phase:,Conditions:[],Addresses:[],DaemonEndpoints:NodeDaemonEndpoints{KubeletEndpoint:DaemonEndpoint{Port:0,},},NodeInfo:NodeSystemInfo{MachineID:,SystemUUID:,BootID:,KernelVersion:,OSImage:,ContainerRuntimeVersion:,KubeletVersion:,KubeProxyVersion:,OperatingSystem:,Architecture:,},Images:[],VolumesInUse:[],VolumesAttached:[],Config:nil,},}": Timeout: request did not complete within requested timeout 30s",
```

trying to fix that

**Comment by runcom** [comment]
damn, aws limit

/retest

**Comment by kikisdeliveryservice** [comment]
I reported the aws limit prob, hopefully there'll be some cleanup shortly.

**Comment by kikisdeliveryservice** [comment]
saw this here as well: https://openshift-gce-devel.appspot.com/build/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-upgrade/811

File: pkg/daemon/daemon.go [hunk 1]
```diff
@@ -845,6 +850,9 @@ func (dn *Daemon) CheckStateOnBoot() error {
 		// All good!
 		return nil
 	}
+	if dn.recorder != nil {
+		dn.recorder.Eventf(getNodeRef(dn.node), corev1.EventTypeNormal, "BootResync", fmt.Sprintf("Booting node %s, currentConfig %s, desiredConfig %s", dn.node.Name, state.currentConfig.GetName(), state.desiredConfig.GetName()))
```
Comment (cgwalters) [inline pkg/daemon/daemon.go]: Note that the name of this function is wrong, it should probably be `CheckStateOnMCDStart` or something, so this message would be `MCD starting on %s ...`

File: pkg/controller/node/status.go [hunk 2]
```diff
@@ -191,7 +195,7 @@ func getUnavailableMachines(currentConfig string, nodes []*corev1.Node) []*corev
 		nodeNotReady := !isNodeReady(node)
 		if dconfig == currentConfig && (dconfig != cconfig || nodeNotReady) {
 			unavail = append(unavail, node)
-			glog.V(2).Infof("Node %s unavailable: different configs %v or node not ready %v", node.Name, dconfig != cconfig, nodeNotReady)
+			glog.V(2).Infof("Node %s unavailable: different configs (desired: %s, current %s) or node not ready %v", node.Name, dconfig, cconfig, nodeNotReady)
```
Comment (cgwalters) [inline pkg/controller/node/status.go]: Probably at this point split into separate conditions with separate logging so we know which happened?

**Pull request merged**